handler/PostMessageSentHandler.ts

```diff
@@ -0,0 +1,65 @@
+import { IHttp, IModify, IPersistence, IRead } from '@rocket.chat/apps-engine/definition/accessors';
+import { IApp } from '@rocket.chat/apps-engine/definition/IApp';
+import { IMessage } from '@rocket.chat/apps-engine/definition/messages';
+import { RoomType } from '@rocket.chat/apps-engine/definition/rooms';
+import { IUser } from '@rocket.chat/apps-engine/definition/users';
+import { AppSettingId } from '../AppSettings';
+import { IParsedDialogflowResponse } from '../definition/IParsedDialogflowResponse';
+import { getAppSetting, getBotUser, getLivechatRoom, getSessionId } from '../helper';
+import { DialogflowSDK } from '../lib/Dialogflow/DialogflowSDK';
+import { AppPersistence } from '../lib/persistence';
+import { SynchronousHandover } from '../lib/SynchronousHandover';
+
+export class PostMessageSentHandler {
+    constructor(private app: IApp,
+                private message: IMessage,
+                private read: IRead,
+                private http: IHttp,
+                private persis: IPersistence,
+                private modify: IModify) {}
+
+    public async run() {
+        const SettingBotUsername: string = await getAppSetting(this.read, AppSettingId.DialogflowBotUsername);
+        if (this.message.sender.username === SettingBotUsername) {
+            // this msg was sent by the Bot itself, so no need to respond back
+            return;
+        } else if (this.message.room.type !== RoomType.LIVE_CHAT) {
+            // check whether this is a Livechat message
+            return;
+        } else if (SettingBotUsername !== getBotUser(this.message).username) {
```

Comment:
You don't need else, you are returning.
```suggestion
        if (SettingBotUsername !== getBotUser(this.message).username) {
```